torchvision/prototype/transforms/_container.py

```diff
@@ -0,0 +1,92 @@
+from typing import Any, List
+
+import torch
+from torch import nn
+from torchvision.prototype.transforms import Transform
+
+
+class ContainerTransform(nn.Module):
+    def supports(self, obj: Any) -> bool:
+        raise NotImplementedError()
+
+    def forward(self, *inputs: Any, strict: bool = False) -> Any:
+        raise NotImplementedError()
+
+    def _make_repr(self, lines: List[str]) -> str:
+        extra_repr = self.extra_repr()
+        if extra_repr:
+            lines = [self.extra_repr(), *lines]
+        head = f"{type(self).__name__}("
+        tail = ")"
+        body = [f"  {line.rstrip()}" for line in lines]
+        return "\n".join([head, *body, tail])
+
+
+class WrapperTransform(ContainerTransform):
+    def __init__(self, transform: Transform):
+        super().__init__()
+        self._transform = transform
+
+    def supports(self, obj: Any) -> bool:
+        return self._transform.supports(obj)
+
+    def __repr__(self) -> str:
+        return self._make_repr(repr(self._transform).splitlines())
+
+
+class MultiTransform(ContainerTransform):
+    def __init__(self, *transforms: Transform) -> None:
+        super().__init__()
+        self._transforms = transforms
+
+    def supports(self, obj: Any, *, strict: bool = False) -> bool:
+        aggregator = all if strict else any
+        return aggregator(transform.supports(obj) for transform in self._transforms)
+
+    def __repr__(self) -> str:
+        lines = []
+        for idx, transform in enumerate(self._transforms):
+            partial_lines = repr(transform).splitlines()
+            lines.append(f"({idx:d}): {partial_lines[0]}")
+            lines.extend(partial_lines[1:])
+        return self._make_repr(lines)
+
+
+class Compose(MultiTransform):
+    def forward(self, *inputs: Any, strict: bool = False) -> Any:
+        sample = inputs if len(inputs) > 1 else inputs[0]
+        for transform in self._transforms:
+            sample = transform(sample, strict=strict)
+        return sample
+
+
+class RandomApply(WrapperTransform):
```

Comment:
Is that fair to assume that the pattern we will want to follow for now on is to instead of implementing `RandomHorizontalFlip` to instead just to `RandomApply(HorizontalFlip)` ?

Reply:
Yes, that is the pattern that I have in mind. That way we would only need to implement the "base" version of each transform without loosing any functionality. Plus, the user now has the option to use `HorizontalFlip` without a random component.

Note that there will be some `RandomX` transforms, namely ones where the parameter sampling has a random component. Since they otherwise behave like the "base" transform, it is sufficient to only implement the sampling and wrap  the "base" version. See `Resize` and `RandomResize` from this PR for an example.